internal/layer2/announcer.go

```diff
@@ -181,6 +224,8 @@ func (a *Announce) shouldAnnounce(ip net.IP) dropReason {
 
 // SetBalancer adds ip to the set of announced addresses.
 func (a *Announce) SetBalancer(name string, ip net.IP) {
+	// Call spamStart at the end of the function without holding the lock
+	defer a.spamStart(name)
```

Comment:
Another alternative to doing this unconditionally every time as this patch does, might be to have `ForceSync()` take a bool `unconditionalAnnounce` param and set it to true when processing the memberlist event. And, of course, spam if that is set. But seems messy to pass that "bool" around on first sight.

Yeah, probably discard this

Reply:
can we resolve this one ?